PopPUNK/models.py

```diff
@@ -991,7 +1000,19 @@ def assign(self, X, slope=None):
 # Wrapper function for LineageFit.__reduce_rank__ to be called by
 # multiprocessing threads
 def reduce_rank(lower_rank, fit, higher_rank_sparse_mat, n_samples, dtype):
-    fit.__reduce_rank__(higher_rank_sparse_mat, lower_rank, n_samples, dtype)
+    # Only modify the matrix if the method or rank differs - otherwise save in unmodified form
+    if lower_rank==fit.max_search_depth and fit.reciprocal_only is False and fit.count_unique_distances is False:
+        fit.__save_sparse__(higher_rank_sparse_mat[2],
```

Comment:
Is saving expected from calling this function? Maybe it would be clearer if it did nothing/raised a message

Reply:
`__reduce_rank__` has always called `__save_sparse__`, the added code just dodges unnecessary processing - it is a superfluous file copy, but keeps database structure simpler